schemas/index.js

```diff
@@ -0,0 +1,142 @@
+const Joi = require("@hapi/joi");
+
+const userSchema = Joi.object().keys({
+  username: Joi.string().alphanum(),
+  email: Joi.string()
+    .email({ minDomainSegments: 2 })
+    .required(),
+  location: Joi.string()
+    .alphanum()
+    .required(),
+  image: Joi.string().allow(null),
+  banner_image: Joi.string().allow(null),
+  bio: Joi.string().allow(null)
+});
+
+const groupSchema = Joi.object().keys({
+  group_name: Joi.string().required(),
+  privacy_setting: Joi.string()
+    .required()
+    .valid("public", "private", "hidden"),
+  location: Joi.string()
+    .alphanum()
+    .required(),
+  creator_id: Joi.number()
+    .integer()
+    .min(1)
+    .required(),
+  image: Joi.string().allow(null)
+});
+
+const allegianceSchema = Joi.object().keys({
+  allegiance_name: Joi.string().required(),
+  image: Joi.string().allow(null),
+  banner_image: Joi.string().allow(null),
+  sport: Joi.string().allow(null)
+});
+
+const groupAllegianceSchema = Joi.object().keys({
+  group_id: Joi.number()
+    .integer()
+    .min(1)
+    .required(),
+  allegiance_id: Joi.number()
+    .integer()
+    .min(1)
+    .required()
+});
+
+const groupUserSchema = Joi.object().keys({
+  user_id: Joi.number()
+    .integer()
+    .min(1)
+    .required(),
+  // Would like to add valid options below once we know what they are
+  user_type: Joi.string().required(),
```

Comment:
Definitely revisit this point once the user "types" get fleshed out, will need to be modified

Reply:
Yep, and there is a good example of using valid() elsewhere in this file.